include/revng/ABIAnalyses/Generated/DeadRegisterArgumentsOfFunction.h

```diff
@@ -0,0 +1,169 @@
+        return LatticeElement::NoOrDead;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    case TransferFunction::Write:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::NoOrDead;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    default:
+      return E;
+    }
+  }
+};
+template<bool isForward>
+struct MFI : ABIAnalyses::ABIAnalysis {
+  using LatticeElement = llvm::DenseMap<const llvm::GlobalVariable *,
+                                        CoreLattice::LatticeElement>;
+  using Label = const llvm::BasicBlock *;
+  using GraphType = std::conditional_t<isForward,
+                                       const llvm::BasicBlock *,
+                                       llvm::Inverse<const llvm::BasicBlock *>>;
+  using GT = llvm::GraphTraits<GraphType>;
+  using LGT = GraphType;
+
+  LatticeElement
+  combineValues(const LatticeElement &Lh, const LatticeElement &Rh) const {
+    return ABIAnalyses::combineValues<LatticeElement, CoreLattice>(Lh, Rh);
+  };
+
+  bool isLessOrEqual(const LatticeElement &Lh, const LatticeElement &Rh) const {
+    return ABIAnalyses::isLessOrEqual<LatticeElement, CoreLattice>(Lh, Rh);
+  };
+
+  LatticeElement applyTransferFunction(Label L, const LatticeElement &E) const {
+    LatticeElement New = E;
+    std::vector<const Instruction *> InsList;
+    for (auto &I : make_range(L->begin(), L->end())) {
+      InsList.push_back(&I);
+    }
+    for (size_t i = 0; i < InsList.size(); i++) {
+      auto I = InsList[isForward ? i : (InsList.size() - i - 1)];
+      TransferKind T = classifyInstruction(I);
+      switch (T) {
+      case TheCall: {
+        for (auto &Reg : getRegisters()) {
+          New[Reg] = CoreLattice::transfer(TheCall,
+                                           getOrDefault<LatticeElement,
+                                                        CoreLattice>(New, Reg));
+        }
+        break;
+      }
+      case Read:
+        for (auto &Reg : getRegistersRead(I)) {
+          New[Reg] = CoreLattice::transfer(T,
+                                           getOrDefault<LatticeElement,
+                                                        CoreLattice>(New, Reg));
+        }
+        break;
+      case WeakWrite:
+      case Write:
+        for (auto &Reg : getRegistersWritten(I)) {
+          New[Reg] = CoreLattice::transfer(T,
+                                           getOrDefault<LatticeElement,
+                                                        CoreLattice>(New, Reg));
+        }
+        break;
+      default:
+        break;
+      }
+    }
+    return New;
+  };
+};
```

Comment:
I don't think we need to generated this.
It can be just a template function/method.

Reply:
moved out of the generated code